### DISCUSSION TITLE
Taking screenshots of GUI window using a custom C++ code

### DISCUSSION THREAD

**Discussion by skype4uts** [Ho-Ro/Hantek6022API#29] [Q&A]
I am working on automating a set of test instruments in our lab for which I need to take screenshots at certain intervals based on inputs from other instruments. I couldn't find  a way on how to do it, I did checked the Hantek6022 Python API by Ho-Ro but that only collects the waveform data and plots it, but OpenHantek also shows the measurement values such as peak to peak voltages, etc, so If I am able to take screenshot by using or extending the required code for it, then I could also get those measurement values. So please help me with how to get it ?

**Comment by Ho-Ro** [comment]
I'm using Qt functions in OpenHantek6022, see the [relevant code](https://github.com/OpenHantek/OpenHantek6022/blob/main/openhantek/src/mainwindow.cpp#L576-L691) and a simple striped down solution: 

```cpp
    ...
    auto activeWindow = qApp->activeWindow();
    QPixmap screenshot( activeWindow->size() ); // prepare a pixmap with the correct size

    QDateTime now = QDateTime::currentDateTime();
    QString fileName = now.toString( tr( "yyyyMMdd_hhmmss" ) );
    fileName += ".png";

    activeWindow->render( &screenshot ); // take the screenshot

    // here we have a screenshot ...
    screenshot.save( fileName );
    ...
```
But if I understand your intent correctly you have different graphical measurement programs running on your desktop and you want to make screenshots of individual windows triggered by external events. This cannot be done easily with my approach unless each program provides a measure for it. You should better use a tool like `import` from the ImageMagick suite or check [this article](https://askubuntu.com/questions/194427/what-is-the-terminal-command-to-take-a-screenshot).

**Reply by skype4uts** [reply]
Thanks for your reply and the pointers. Yes as you said latter, I need to take the screenshot of window of OpenHantek GUI based on external events.  I made it by using pywin32api , which gets the bitmap of window and it is processed and saved into a png file (even if the window remains inactive). 

Also Sorry as I intended to ask this in OpenHantek discussions instead of Hantek6022API  and mistakenly asked here anyway if this API also has that support it would be nice too.
